store/store.go

```diff
@@ -300,26 +302,41 @@ func (bs *BlockStore) PruneBlocks(height int64) (uint64, error) {
 		return nil
 	}
 
+	evidencePoint := height
 	for h := base; h < height; h++ {
+
 		meta := bs.LoadBlockMeta(h)
 		if meta == nil { // assume already deleted
 			continue
 		}
-		if err := batch.Delete(calcBlockMetaKey(h)); err != nil {
-			return 0, err
+
+		// This logic is in place to protect data that proves malicious behavior.
+		// Instead of pruning all blocks we only prune the meta and commit data
+		// if the evidence has exceeded the block height.
+
+		if evidencePoint == height && !evidence.IsEvidenceExpired(state.LastBlockHeight, state.LastBlockTime, h, meta.Header.Time, state.ConsensusParams.Evidence) {
+			evidencePoint = h
```

Comment:
Could you not just break here? Once you discover the evidence point you do not delete anything except flush the existing batch? And line 326  `batch.Delete(calcBlockHashKey(meta.BlockID.Hash))` should not execute at all, same for line 334 (`err := batch.Delete(calcSeenCommitKey(h))`)? For the code on these lines tehre may be context I am missing so maybe I am wrong here.

Reply:
Once we pass the evidence point, we are still pruning the entire block and the seen commit

Reply:
This data is not needed for evidence verification

Reply:
This may have been confusing with my comments coupled wit the logic implemented

Reply:
have added some more comments which might shed some light

Reply:
Thanks for the clarification, makes sense.